x-pack/plugins/observability/public/pages/rule_details/components/rule_detail_tabs.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useRef, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTabbedContent,
+  EuiTabbedContentTab,
+} from '@elastic/eui';
+import { AlertConsumers } from '@kbn/rule-data-utils';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Query, BoolQuery } from '@kbn/es-query';
+import { Rule, RuleType } from '@kbn/triggers-actions-ui-plugin/public';
+import { RuleTypeParams } from '@kbn/alerting-plugin/common';
+import { ALERTS_FEATURE_ID } from '@kbn/alerting-plugin/common';
+
+import { ObservabilityAppServices } from '../../../application/types';
+import { fromQuery, toQuery } from '../../../utils/url';
+import { ObservabilityAlertSearchbarWithUrlSync } from '../../../components/shared/alert_search_bar';
+import {
+  EXECUTION_TAB,
+  ALERTS_TAB,
+  RULE_DETAILS_PAGE_ID,
+  RULE_DETAILS_ALERTS_SEARCH_BAR_ID,
+  URL_STORAGE_KEY,
+} from '../constants';
+import { TabId } from '../types';
+import { observabilityFeatureId } from '../../../../common';
+
+interface TabProps {
+  rule: Rule<RuleTypeParams>;
+  ruleType: RuleType<string, string> | undefined;
+}
+
+export function RuleDetailTabs({ rule, ruleType }: TabProps) {
+  const {
+    triggersActionsUi: {
+      alertsTableConfigurationRegistry,
+      getRuleEventLogList: RuleEventLogList,
+      getAlertsStateTable: AlertsStateTable,
+    },
+  } = useKibana<ObservabilityAppServices>().services;
+
+  const history = useHistory();
+  const location = useLocation();
+
+  const ruleQuery = useRef([
+    { query: `kibana.alert.rule.uuid: ${rule.id}`, language: 'kuery' },
+  ] as Query[]);
+
+  const [esQuery, setEsQuery] = useState<{ bool: BoolQuery }>();
+
+  const [tabId, setTabId] = useState<TabId>(
+    (toQuery(location.search)?.tabId as TabId) || EXECUTION_TAB
+  );
+
+  const updateUrl = (nextQuery: { tabId: TabId }) => {
+    history.push({
+      ...location,
+      search: fromQuery({
+        ...toQuery(location.search),
+        ...nextQuery,
+      }),
+    });
+  };
+
+  const onTabIdChange = (newTabId: TabId) => {
+    setTabId(newTabId);
+    updateUrl({ tabId: newTabId });
+  };
+
+  const features = (
+    rule?.consumer === ALERTS_FEATURE_ID && ruleType?.producer ? ruleType.producer : rule?.consumer
```

Comment:
This to me looks slightly magical, but it is unchanged from the previous implementation. Perhaps someone can illuminate this variable.